File: java/src/hackerrank/BreakingBestAndWorstRecords.java
package hackerrank;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

//https://www.hackerrank.com/challenges/breaking-best-and-worst-records/problem
public class BreakingBestAndWorstRecords {

	public static int[] breakingRecords(int[] scores) {
		int countMin = 0, countMax = 0;
		int min = scores[0], max = scores[0];
		for (int score : scores) {
			if (score < min) {
				min = score;
				countMin++;
			} else if (score > max) {
				max = score;
				countMax++;
			}
		}
		return new int[]{countMax, countMin};
	}


	@Test
	void testBreakingRecords() {
		int[] sample0 = {10, 5, 20, 20, 4, 5, 2, 25, 1};
		int[] result = breakingRecords(sample0);
		assertEquals(2, result[0]);
		assertEquals(4, result[1]);
		int[] sample1 = {3, 4, 21, 36, 10, 28, 35, 5, 24, 42};
		result = breakingRecords(sample1);
		assertEquals(4, result[0]);
		assertEquals(0, result[1]);		
	}
	
}

File: javascript/hackerrank/breaking-best-and-worst-records.js
// https://www.hackerrank.com/challenges/breaking-best-and-worst-records/problem

const breakingRecords = (scores) => {
    let countMin = 0, countMax = 0;
    let min = scores[0], max = scores[0];
    
    for (let i = 0; i < scores.length; i++) {
        if (scores[i] < min) {
            min = scores[i];
            countMin++;
        } else if (scores[i] > max) {
            max = scores[i];
            countMax++;
        }
    }
    
    const result = [countMax, countMin];
    return result;
}

test("sample 0", () => {
    const sample0 = [10, 5, 20, 20, 4, 5, 2, 25, 1];
	const result = breakingRecords(sample0);
	expect(result[0]).toEqual(2);
	expect(result[1]).toEqual(4);
});

test("sample 1", () => {
    const sample1 = [3, 4, 21, 36, 10, 28, 35, 5, 24, 42];
	const result = breakingRecords(sample1);
	expect(result[0]).toEqual(4);
	expect(result[1]).toEqual(0);
});
